src/log-store/src/kafka/index/collector.rs

```diff
@@ -148,16 +158,69 @@ impl GlobalIndexCollector {
         let task = CollectionTask {
             providers: providers.clone(),
             dump_index_interval,
-            operator,
+            operator: operator.clone(),
             path,
             running: Arc::new(AtomicBool::new(true)),
         };
-        task.run();
-        Self { providers, task }
+        let handle = task.run();
+        Self {
+            providers,
+            operator,
+            _handle: handle,
+        }
+    }
+
+    #[cfg(test)]
+    pub fn new_for_test(operator: object_store::ObjectStore) -> Self {
+        Self {
+            providers: Default::default(),
+            operator,
+            _handle: Default::default(),
+        }
     }
 }
 
 impl GlobalIndexCollector {
+    /// Retrieve [`EntryId`]s for a specified `region_id` in `datanode_id`
+    /// that are greater than or equal to a given `entry_id`.
+    pub(crate) async fn read_remote_region_index(
+        &self,
+        datanode_id: u64,
+        provider: &KafkaProvider,
+        region_id: RegionId,
+        entry_id: EntryId,
+    ) -> Result<Option<(BTreeSet<EntryId>, EntryId)>> {
+        let path = default_index_file(datanode_id);
+        let exists = self
+            .operator
+            .is_exist(&path)
```

Comment:
How about `read` the data directly? This way, there will be one less call.

Reply:
Resolved in [6a65cbf](https://github.com/GreptimeTeam/greptimedb/pull/4565/commits/6a65cbf5fab7658196c4cdba4ebcbecbcff38f92)